graphql_compiler/cost_estimation/cardinality_estimator.py

```diff
@@ -69,17 +69,103 @@ def _get_last_edge_direction_and_name_to_location(location):
     return edge_direction, edge_name
 
 
-def _get_parent_and_child_name_from_edge(schema_graph, child_location):
+def _get_base_classnames_of_parent_and_child_from_edge(schema_graph, child_location):
     """Get the base classname of a location and its parent from the last edge information."""
     edge_direction, edge_name = _get_last_edge_direction_and_name_to_location(child_location)
     edge_element = schema_graph.get_edge_schema_element_or_raise(edge_name)
     if edge_direction == INBOUND_EDGE_DIRECTION:
-        parent_name_from_edge = edge_element.base_out_connection
-        child_name_from_edge = edge_element.base_in_connection
+        parent_base_classname = edge_element.base_out_connection
+        child_base_classname = edge_element.base_in_connection
     elif edge_direction == OUTBOUND_EDGE_DIRECTION:
-        parent_name_from_edge = edge_element.base_in_connection
-        child_name_from_edge = edge_element.base_out_connection
-    return parent_name_from_edge, child_name_from_edge
+        parent_base_classname = edge_element.base_in_connection
+        child_base_classname = edge_element.base_out_connection
+    else:
+        raise AssertionError(u'Expected edge direction to be either inbound or outbound.'
+                             u'Found: edge {} with direction {}'.format(edge_name, edge_direction))
+    return parent_base_classname, child_base_classname
+
+
+def _get_outbound_and_inbound_names_from_locations(query_metadata, child_location, parent_location):
+    """Get the out and in classnames of a location and its parent from the last edge information."""
+    edge_direction, edge_name = _get_last_edge_direction_and_name_to_location(child_location)
+    parent_name_from_location = query_metadata.get_location_info(parent_location).type.name
+    child_name_from_location = query_metadata.get_location_info(child_location).type.name
+    if edge_direction == INBOUND_EDGE_DIRECTION:
+        outbound_vertex_name = child_name_from_location
+        inbound_vertex_name = parent_name_from_location
+    elif edge_direction == OUTBOUND_EDGE_DIRECTION:
+        outbound_vertex_name = parent_name_from_location
+        inbound_vertex_name = child_name_from_location
+    else:
+        raise AssertionError(u'Expected edge direction to be either inbound or outbound.'
+                             u'Found: edge {} with direction {}'.format(edge_name, edge_direction))
+    return outbound_vertex_name, inbound_vertex_name
+
+
+def _probe_statistics_for_vertex_edge_vertex_count(
+    statistics, query_metadata, child_location, parent_location
+):
+    """Probe for the count of edges that connect parent_location and child_location vertices."""
```

Comment:
What does `probe` mean here? It feels a bit vague, and the docstring uses the same word so it's hard to figure out what this function actually does.

Reply:
Decided on `query` instead of `probe`.